src/Microsoft.IdentityModel.Xml/KeyInfo.cs

```diff
@@ -145,6 +145,9 @@ public override int GetHashCode()
         /// </summary>
         internal bool MatchesKey(SecurityKey key)
         {
+            if (key == null)
+                return false;
+
             if (key is X509SecurityKey x509SecurityKey)
```

Comment:
not sure if this line throws null ref exception. That's why adding code to return null if key is null.